feat(curriculum): add workshop python functions (#68279)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-kitchen-inventory-tracker/6a3e5ccb99fd6053af9032a1.md

---
id: 6a3e5ccb99fd6053af9032a1
title: Step 6
challengeType: 20
dashedName: step-6
---

# --description--

> Now create another function. This time, your function will take "parameters". Parameters are like placeholders for data that you pass into the function when you call it.

Reviewer: We should have an example over here on how to use parameters.

> Below your `take_inventory` function define an empty function named `use_eggs` that accepts two parameters: `available_eggs` and `eggs_to_use`, in that order. 

Reviewer: ```suggestion
Below your `take_inventory` function, define an empty function named `use_eggs` that accepts two parameters: `available_eggs` and `eggs_to_use`, in that order. 
```

> This function will also introduce a concept called **name shadowing**. If you name a parameter the exact same name as a global variable (like `eggs`), the function will use its own local parameter `eggs` instead of the global one. The local variable "shadows" the global one!

Reviewer 2: this is not actually true anymore, name shadowing does not happen with this function with the new names

now, the new parameter names make much more sense, but name shadowing does not happen anymore

so need to consider if name shadowing is an essential part of this workshop (it still happens with `make_friend_egg`, even if there is no changing of `eggs` so it's not the same)

Author: i have two option change available_eggs name to eggs or remove the name shadowing description line.
what do you think ?

Author: i have decided no to introduce name shadowing concept in this workshop , may be we can introduce it in another workshop.

> This function will also introduce a concept called **name shadowing**. If you name a parameter the exact same name as a global variable (like `available_eggs`), the function will use its own local parameter `available_eggs` instead of the global one. The local variable "shadows" the global one!

Reviewer: ```suggestion
This function will also introduce a concept called **name shadowing**. If you give a parameter the exact same name as a global variable (like `available_eggs`), the function will use its own local parameter `available_eggs` instead of the global one. The local variable "shadows" the global one!
```

Use the `pass` keyword inside the body for now.

# --hints--

You should define a function named `use_eggs`.

```js
({
  test: () => assert(
    runPython(`_Node(_code).find_function("use_eggs")`)
  )
})
```

> Your use_eggs function should have eggs and num_eggs as its parameters.

Reviewer 2: ```suggestion
Your `use_eggs` function should have `eggs` and `num_eggs` as its parameters.
```

---
id: 6a3e5ccb99fd6053af9032a1
title: Step 6
challengeType: 20
dashedName: step-6
---

# --description--

> Now that you know what's in your kitchen, you can use those items to cook something. To do this, create a new function that accepts parameters. Parameters act as placeholders for the data you pass into a function when you call it.

Reviewer: ```suggestion
Now that you know what's in your kitchen, you can use those items to cook. To do this, create a new function that accepts parameters. Think of parameters as placeholders for the data you pass into the function when calling it.
```

> For example:

Reviewer: ```suggestion
Here is a function that accepts two parameters, `a` and `b`, and prints their sum to the terminal.
```

```py

> def my_function(parameter1, parameter2):

Reviewer: ```suggestion
def calculate_sum(a, b):
    print(a + b)

calculate_sum(3, 1) # 4
```

    print(a + b)

calculate_sum(3, 1) # Output: 4  
```

Below your `check_kitchen_stock` function, define an empty function named `use_eggs` that accepts two parameters: `available_eggs` and `eggs_to_use`, in that order. 

This function will also introduce a concept called **name shadowing**. If you give a parameter the exact same name as a global variable (like `available_eggs`), the function will use its own local parameter `available_eggs` instead of the global one. The local variable "shadows" the global one!

Use the `pass` keyword inside the body for now.

# --hints--

You should define a function named `use_eggs`.

```js
({
  test: () => assert(

>     runPython(`_Node(_code).find_function("use_eggs")`)

Reviewer 2: ```suggestion
    runPython(`_Node(_code).has_function("use_eggs")`)
```
